src/DocumentDB.ChangeFeedProcessor/Exceptions/LeaseLostException.cs

```diff
@@ -8,46 +8,86 @@ namespace Microsoft.Azure.Documents.ChangeFeedProcessor.Exceptions
     using System.Runtime.Serialization;
     using Microsoft.Azure.Documents.ChangeFeedProcessor.PartitionManagement;
 
+    /// <summary>
+    /// Exception occurred when lease was taken by a different host.
+    /// </summary>
     [Serializable]
-    internal class LeaseLostException : Exception
```

Comment:
We should derive all these from something like ChangeFeedProcessorException for easier filtering.

Reply:
Could you please describe the proposal in a separate issue/PR, so we can understand the issue and decide what approach would be feasible to solve it?

Custom common exception might be a good idea, but let's move the related discussion out of the scope of this change.

Reply:
For this case though it should be OK to introduce base class as separate change.